perf/top_address.go

```diff
@@ -0,0 +1,113 @@
+package perf
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/prque"
+	"github.com/gin-gonic/gin"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+var addrStatsEnabled, _ = getEnvBool("METRICS_TOP_ADDRESS_ENABLED")
+
+type AddrTxStats struct {
+	Map   map[string]int64 `json:"map"`
+	Start time.Time        `json:"start"`
+	End   time.Time        `json:"end"`
+}
+
+var eoaMap = make(map[string]int64)
+var contractMap = make(map[string]int64)
+
+var start = time.Now()
+
+func UpdateTopEOAStats(address common.Address) {
+	if !addrStatsEnabled {
+		return
+	}
+	inc(eoaMap, address.Hex())
+}
+
+func UpdateTopContractStats(address common.Address) {
+	if !addrStatsEnabled {
+		return
+	}
+	inc(contractMap, address.Hex())
+}
+
+func inc(m map[string]int64, address string) {
```

Comment:
I think we should limit the size of the map, if we want top1K, the map should limit at 10K size. Otherwise, OOM will happen.

Reply:
updated, use a ticker to clean the maps